### ISSUE TITLE
Y-axis scaling error in Scree Plots for factor analysis

### ISSUE THREAD

**Issue by gbg33k** [jasp-stats/jasp-issues#556]
* JASP version: 0.11.1

* OS name and version: Windows
* Bug description:  Scree Plot for factor analysis has incorrect scaling on vertical axis.  For a data set that has the eigenvalues of 3.3535, 1.6674, 1.5860, 0.1879, 0.1104, 0.0506, 0.0442, the scree plot should look like the first image below (correctly generated by JASP 0.9.2.0).  However, in later versions (I first noticed the problem in version 0.10.2.0), the vertical scale is incorrect - see second image below.

CORRECT SCALE ON Y-AXIS (JASP 0.9.2.0)
![image](https://user-images.githubusercontent.com/57960628/69173591-0fa16e80-0ace-11ea-8426-be3a572be5b6.png)

INCORRECT SCALE ON Y-AXIS (JASP 0.11.1)
![image](https://user-images.githubusercontent.com/57960628/69173416-bb968a00-0acd-11ea-925b-01918758509a.png)

**Comment by vankesteren** [comment]
Hi, 

it seems from your images that this issue has to do with the difference between factor analysis and principal components analysis. Here's what the `psych` package author has to say about it (from [this link](https://www.rdocumentation.org/packages/psych/versions/1.8.12/topics/fa.parallel)):

> Parallel analysis for factors is actually harder than it seems, for the question is what are the appropriate communalities to use. [...] By default, fa.parallel estimates the communalities based upon a one factor minres solution. Although this will underestimate the communalities, it does seem to lead to better solutions on simulated or real (e.g., the bfi or Harman74) data sets.

So perhaps, the y-axis for the factor analysis should not be labelled as "eigenvalues" but rather as "communalities". What do you think?

**Comment by gbg33k** [comment]
Hello!

Thanks for your email.  The option to turn on that plot is titled “scree plot”.  Scree plots should plot eigenvalues and not communalities.  Note that horizontal line appears at the value which corresponds to the value that the user enters for  “Eigenvalues > ___”.  If the y axis is relabelled to communalities, then both the “scree plot” and “eigenvalues > ____” options would not make any sense.

Regarding your point about exploratory factor analysis vs principal components analysis, I performed an exploratory factor analysis in both cases.  Neither output came from a PCA.  I know that the older version labels the horizontal axis as “components” but that is actually an error in the earlier version of JASP. That error has since been fixed in later versions, and the axis is now labelled as “factors”.

In summary, the scree plot should show eigenvalues and not communalities.  If the issue did not result from incorrect scaling, then the code for determining the eigenvalues needs correction.  It may be the case that parallel analysis requires a separate plot which shows communalities.

Thank you!
Christine



Sent from my iPhone
________________________________

**Comment by gbg33k** [comment]
Hello again,

I re-examined the data and I do think the issue is a plotting error and not an issue with the FA package.  I double-checked the analysis in R (console output below).  The eigenvalues generated by the parallel analysis are highlighted in green below, which do not match with the values plotted in JASP (see screenshot after the output please).  In case it is of any help, here is the [data](https://github.com/jasp-stats/jasp-issues/files/3897923/data.xlsx) for your testing purposes.  I would appreciate if this data set is not shared outside this forum.  thank you!



Christine

![image](https://user-images.githubusercontent.com/57960628/69743218-12bedf00-110c-11ea-9b49-447cc7c784e5.png)


![image](https://user-images.githubusercontent.com/57960628/69742943-95936a00-110b-11ea-8326-4494c8d5ea59.png)

**Comment by vankesteren** [comment]
Hi Christine,

in JASP we use the `psych` package directly to perform parallel analysis for this scree plot (see [this line of code](https://github.com/jasp-stats/jasp-desktop/blob/fa815f41a4074287c40c6fba392bb5e79d63bd9e/JASP-Engine/JASP/R/exploratoryfactoranalysis.R#L390)). The difference is that what is called `eigenvalue` is actually not a pure eigenvalue, but "estimated communalities based upon a one factor minres solution". 

So it's a semantic difference. The `psych` package author says that his method (and thus, the method JASP uses) leads to better results.

I am in favour of renaming the y axis to "communalities", though.

**Comment by gbg33k** [comment]
Hello, I'm sorry for the slow response.  Ok, I support renaming the Y axis to communalities for the current plot, however, the option currently labelled "Eigenvalue > 1" will need to be updated as well.   Furthermore, is it possible to provide the true Eigenvalues in a table as an optional output?
